### PR TITLE
Gradients Refactor

### PR DESCRIPTION
**Pull request by joebell93**
## Description
- Re-define our `$gradients` map to use percentage color stops
- All gradient tools moved to `tools/gradients`
- New `gradient-color-stops()` function
- `gradient()` function renamed to `gradient-color()`
- `@include gradient-background()` amended to support new stops
- Remove `@include page-background()` mixin
- Renamed keys in `$gradients` (see CHANGELOG.md)

## Related  Issue
https://github.com/sky-uk/toolkit/issues/223

## Motivation and Context
Toolkit 2.0 progression

## How Has This Been Tested?
Tested in UI (PR incoming)

## Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [x] All code includes full inline documentation (as per the [template](https://github.com/sky-uk/toolkit-core/blob/master/_template.scss)).
- [x] All code conforms to the Toolkit [coding style](https://github.com/sky-uk/toolkit/wiki/Coding-Style).
- [x] All code conforms to [WCAG 2.0 level AA Accessibility Guidelines](https://www.w3.org/TR/WCAG20/).
- [x] New functions and mixins have appropriate tests.
- [x] All new and existing tests passed.
- [x] I have added instructions on how to test my changes.
- [ ] Package version updated.
- [x] CHANGELOG.md updated.

### CODE REVIEW

**Comment by MrDinsdale** [comment]
Added a proposal to tackle this. This allows us to loop through values in a gradient without relying on hardcoded values. Should be easy to integrate into `toolkit-ui` too.

https://github.com/sky-uk/toolkit-core/pull/193/files

File: settings/_breakpoints.scss [hunk 1]
```diff
@@ -10,7 +10,7 @@
 // which that breakpoint kicks in. These pixel values will be conveted into ems
```
Comment (skitson) [inline settings/_breakpoints.scss]: Typo 'conve(r)ted'.

File: tools/_functions.scss [hunk 2]
```diff
@@ -134,3 +117,22 @@
...
 }
+
+// Reverse list
+// ===========================================
+
+// Function to reverse list of values
+//
+// Example:
+//    reverse(1,2,3,4) => 4,3,2,1
+//    reverst(a 1, b 2, c 3) => c 3, b 2, a 1
```
Comment (skitson) [inline tools/_functions.scss]: Typo 'reverst`.

File: settings/_breakpoints.scss (hunk 1 above)
Comment (joebell93) [inline settings/_breakpoints.scss]: Cheers!

File: tools/_functions.scss (hunk 2 above)
Comment (joebell93) [inline tools/_functions.scss]: 👀

**Pull request merged**